### PR TITLE
Implement "hie" files for ghc-8.6.5

### PR DESCRIPTION
**Pull request by mpickering**
This means that the .hi files patch can also be used with 8.6.5

### CODE REVIEW

**Comment by pepeiborra** [comment]
Thanks @mpickering ! Would you mind targeting pepeiborra:interface-files-0.1.0.0-rebased-again instead?

**Comment by mpickering** [comment]
Done.

File: src/Development/IDE/GHC/Compat.hs [hunk 1]
```diff
@@ -182,3 +171,50 @@ dontWriteHieFiles d =
 #else
     d
 #endif
+
+nameListFromAvails :: [AvailInfo] -> [(SrcSpan, Name)]
+nameListFromAvails as =
+  map (\n -> (nameSrcSpan n, n)) (concatMap availNames as)
+
+#if !MIN_GHC_API_VERSION(8,8,0)
+-- Reimplementations of functions for HIE files for 8.6.5
```
Comment (pepeiborra) [inline src/Development/IDE/GHC/Compat.hs]: Can this comment be relaxed to include older versions of GHC?

Comment (mpickering) [reply src/Development/IDE/GHC/Compat.hs]: I only tested with 8.6.5 but I imagine it works for most recent version.

Comment (pepeiborra) [reply src/Development/IDE/GHC/Compat.hs]: I mean, will it work for GHC 8.4.x? ghcide still supports those versions, so if it doesn't work we would need to keep `supportsHieFiles` around

**Comment by pepeiborra** [comment]
Looks good to me!

File: src/Development/IDE/GHC/Compat.hs (hunk 1 above)
Comment (mpickering) [reply src/Development/IDE/GHC/Compat.hs]: It should work but I didn't test it.

**Pull request merged**